data/modules/CargoRun/CargoRun.lua

```diff
@@ -0,0 +1,1138 @@
+		for i = 1, #escort_ships do
+			if escort_ships[i] == ship then
+				table.remove(escort_ships, i)
+				if isPirateShip(attacker) then
+					attacker:AIKill(Game.player)
+				end
+			end
+		end
+	elseif isPirateShip(ship) then
+		for i = 1, #pirate_ships do
+			if pirate_ships[i] == ship then
+				table.remove(pirate_ships, i)
+				if isEscortShip(attacker) then
+					Comms.ImportantMessage(l.TARGET_DESTROYED, attacker.label)
+					if #pirate_ships ~= 0 then
+						attacker:AIKill(pirate_ships[1])
+						escort_switch_target = Game.time + Engine.rand:Integer(90, 120)
+					end
+				end
+			end
+		end
+	end
+end
+
+local onShipFiring = function (ship)
+	if ship:IsPlayer() then return end
+
+	if isEscortShip(ship) and Game.time >= escort_chatter_time then -- don't flood the control panel with messages
+		Comms.ImportantMessage(l.GUNS_GUNS_GUNS, ship.label)
+			escort_chatter_time = Game.time + Engine.rand:Integer(15, 45)
+	end
+end
+
+local onShipHit = function (ship, attacker)
+	if attacker == nil then return end
+	if not attacker:isa('Ship') then return end
+
+	if ship:IsPlayer() and #escort_ships ~= 0 and not isEscortShip(attacker) then
+		if Game.time >= escort_switch_target then -- don't switch between targets too quick
+			for i = 1, #escort_ships do
+				escort_ships[i]:AIKill(attacker) -- all escort ships respond
+			end
+			Comms.ImportantMessage(l.I_AM_ON_MY_WAY, escort_ships[1].label)
+			escort_switch_target = Game.time + Engine.rand:Integer(90, 120)
+		end
+
+	elseif isEscortShip(ship) then
+		if attacker:IsPlayer() then
+			Comms.ImportantMessage(l.CEASE_FIRE, ship.label)
+		elseif isPirateShip(attacker) then
+			if Game.time >= escort_switch_target then
+				ship:AIKill(attacker)
+				escort_switch_target = Game.time + Engine.rand:Integer(90, 120)
+			end
+			if Game.time >= escort_chatter_time then
+				Comms.ImportantMessage(l.I_NEED_HELP, ship.label)
+				escort_chatter_time = Game.time + Engine.rand:Integer(15, 45)
+			end
+		end
+
+	elseif isPirateShip(ship) then
+		if attacker:IsPlayer() then
+			if Game.time >= pirate_gripes_time then -- don't flood the control panel with messages
+				Comms.ImportantMessage(l["PIRATE_GRIPES_" .. Engine.rand:Integer(1, getNumberOf("PIRATE_GRIPES"))], ship.label)
+				pirate_gripes_time = Game.time + Engine.rand:Integer(30, 90)
+			end
+		elseif isEscortShip(attacker) then
+			if Game.time >= pirate_switch_target then
+				ship:AIKill(attacker)
+				pirate_switch_target = Game.time + Engine.rand:Integer(90, 120)
+			end
+		end
+	end
+end
+
+local onLeaveSystem = function (ship)
+	if ship:IsPlayer() then
+		nearbysystems = nil
+		pirate_ships = {}
+		escort_ships = {}
+	end
+end
+
+local unloadCargo = function (mission)
+	local uc = Game.player:RemoveEquip(mission.cargotype, mission.cargo, "cargo")
+	if uc ~= mission.cargo then -- this is ugly
+		local list = Game.player:GetEquip("cargo")
+		for _, i in pairs(list) do
+			if i.l10n_key == mission.cargotype.l10n_key then
```

Comment:
I'd say "amount", since mass is intrinsics to the cargo type.

(so far it is 1t for everyone, but there's a reason I've set it explicitly).

@impaktor is right on the fact that you should store a reference to the actual table instead of its name. And/or (I'd advise the "and" here), you should store all the equipment tables you generated into the savegame, and restore them on load (see data/libs/Equipment.lua for an example), so that you are sure there are no duplicates of the same equip type.

(Yeah, equipment was my pet project for a while, I know it inside and out, hence the nitpicks :-) )